modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteAbstractDynamicCacheStartFailTest.java

```diff
@@ -772,4 +934,254 @@ public BrokenStoreFactory(boolean eOnAllNodes, String gridName) {
+            caches.clear();
+            nodes.clear();
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance registerMBean(Object object, ObjectName name)
+            throws InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException {
+            String node = name.getKeyProperty("igniteInstanceName");
+
+            if (nodes.contains(node) && caches.contains(name.getKeyProperty("group")))
+                throw new MBeanRegistrationException(new Exception("Simulate exception [node=" + node + ']'));
+
+            return origin.registerMBean(object, name);
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance createMBean(String className, ObjectName name)
+            throws ReflectionException, InstanceAlreadyExistsException, MBeanRegistrationException,
+            MBeanException, NotCompliantMBeanException {
+            return origin.createMBean(className, name);
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance createMBean(String className, ObjectName name, ObjectName loaderName)
+            throws ReflectionException, InstanceAlreadyExistsException, MBeanRegistrationException,
+            MBeanException, NotCompliantMBeanException, InstanceNotFoundException {
+            return origin.createMBean(className, name, loaderName);
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance createMBean(String className, ObjectName name, Object[] params, String[] signature)
+            throws ReflectionException, InstanceAlreadyExistsException, MBeanRegistrationException,
+            MBeanException, NotCompliantMBeanException {
+            return origin.createMBean(className, name, params, signature);
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance createMBean(String className, ObjectName name,
+            ObjectName loaderName, Object[] params,
+            String[] signature) throws ReflectionException, InstanceAlreadyExistsException, MBeanRegistrationException,
+            MBeanException, NotCompliantMBeanException, InstanceNotFoundException {
+            return origin.createMBean(className, name, loaderName, params, signature);
+        }
+
+        /** {@inheritDoc} */
+        @Override public void unregisterMBean(ObjectName name) throws InstanceNotFoundException, MBeanRegistrationException {
+            origin.unregisterMBean(name);
+        }
+
+        /** {@inheritDoc} */
+        @Override public ObjectInstance getObjectInstance(ObjectName name) throws InstanceNotFoundException {
+            return origin.getObjectInstance(name);
+        }
+
+        /** {@inheritDoc} */
+        @Override public Set<ObjectInstance> queryMBeans(ObjectName name,
+            QueryExp query) {
+            return origin.queryMBeans(name, query);
+        }
+
+        /** {@inheritDoc} */
+        @Override public Set<ObjectName> queryNames(ObjectName name,

```

Comment:
Second line isn't needed for this method.

Reply:
done